src/transformers/models/switch/modeling_switch.py

```diff
@@ -412,10 +377,16 @@ def forward(self, inputs: torch.Tensor):
 
         # mask out token that don't fit within expert capacity
         expert_gate *= expert_mask_flat
-
-        # combine_tensor: (n_cores, n_tokens, n_experts, expert_capacity)
-        combine_tensor = expert_gate.reshape(num_cores, tokens_per_core, 1, 1) * expert_mask_flat.reshape(num_cores, tokens_per_core, 1, 1) * F.one_hot(expert_index, num_classes=self.config.n_experts).unsqueeze(3) * F.one_hot(position_in_expert, num_classes=expert_capacity)
+        combine_tensor = expert_gate.reshape(num_cores, tokens_per_core, 1, 1) * expert_mask_flat.reshape(num_cores, tokens_per_core, 1, 1) * F.one_hot(expert_index, num_classes=self.config.n_experts).unsqueeze(3) * F.one_hot(position_in_expert, num_classes=expert_capacity+1)
```

Comment:
Interesting - why the +1? I actually got an error locally working on this recently. Curious what your view on the problem there is.

Reply:
Basically, the number of classes are getting small compared to the shape, `+1` is a way to fix, or we can manipulate `expert_capacity` too, I found this to be more straight forward for the logic. 

Quick notebook with some shapes: https://colab.research.google.com/drive/1I3vUduPP0pjLS6aOJsIpjBFl1DJC8zbl?usp=sharing